catalog-scripts/idp-catalog-wizard-bitbucket.py

```diff
@@ -0,0 +1,175 @@
+import argparse
+import requests
+import os
+from requests.auth import HTTPBasicAuth
+import argparse
+from requests.packages.urllib3.util.retry import Retry
+from requests.adapters import HTTPAdapter
+import subprocess
+
+current_directory = os.path.basename(os.getcwd())
+branch  = "main"
+
+yaml_content_template = """
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  tags:
+    - auto-generated
+    - auto
+  name: <repo_name>
+  annotations:
+    backstage.io/source-location: url:<repo_path>
+spec:
+  type: service
+  domian: <workspace_name>
+  system: <project>
+  service: <repo_name>
+  lifecycle: experimental
+  owner: Harness_Account_All_Users
+"""
+
+prefix_path = "services"
+page_size = 50
+def list_repositories(workspace, project_key, username, app_password, repo_pattern=None):
+        
+    page = 1
+    names = []
+    if(project_key == None):
+        url = f"https://api.bitbucket.org/2.0/repositories/{workspace}"
+    else:
+        url = f"https://api.bitbucket.org/2.0/repositories/{workspace}?q=project.key%3D%22{project_key}%22"
+
+    while True:
+
+        if(project_key == None):
+            paginated_url = f'{url}?page={page}&pagelen={page_size}'
+        else:
+            paginated_url = f'{url}&page={page}&pagelen={page_size}'
+
+        response = requests.get(paginated_url, auth=HTTPBasicAuth(username, app_password))
+        if response.status_code == 200:
+            repositories = response.json()['values']
+
+            for entry in repositories:
+                if entry['name'] != current_directory:
+                    names.append(entry['name'])
+            if 'next' in response.json():
+                page += 1
+            else:
+                break
+        else:
+            print(f"Error: {response.json()}")
+            return None
+
+    for name in names:
+        name = name.lower()
+        directory = f"services/{name}"
+        directory_path = os.path.join(prefix_path, name)
+        file_path = os.path.join(f"{directory_path}/", "catalog-info.yaml")
+
+        os.makedirs(directory_path, exist_ok=True)
+
+        updated_yaml_content = yaml_content_template.replace("<repo_name>", name)
+        updated_yaml_content = updated_yaml_content.replace("<repo_path>", f"https://bitbucket.org/{workspace}/{current_directory}/src/{branch}/{directory}/catalog-info.yaml")
+        updated_yaml_content = updated_yaml_content.replace("<workspace_name>", workspace)
+        if project_key is not None:
+            updated_yaml_content = updated_yaml_content.replace("<project>", project_key.lower())
+
+        with open(file_path, "w") as file:
+            file.write(updated_yaml_content)
+
+def register_yamls(workspace, account, x_api_key):
+    
+    print("Registering YAML files...")
+    count = 0
+    api_url = f"https://backstage.qa.harness.io/{account}/idp/api/catalog/locations"
```

Comment:
Can we update this to the prod URLs so that it works for deluxe without changing anything?